packages/eas-cli/src/update/queries.ts

```diff
@@ -2,84 +2,180 @@ import chalk from 'chalk';
+      promptOptions: {
+        title: 'Load more branches?',
+        renderListItems: updateGroup =>
+          renderUpdateGroupsAsTableWithBranchColumn(updateGroup, paginatedQueryOptions),
+      },
+    });
+  }
+}
+
+export async function listAndRenderUpdateGroupsOnBranchAsync({
+  projectId,
+  branchName,
+  paginatedQueryOptions,
+}: {
+  projectId: string;
+  branchName: string;
+  paginatedQueryOptions: PaginatedQueryOptions;
+}): Promise<void> {
+  if (paginatedQueryOptions.nonInteractive) {
+    const updateGroups = await queryUpdateGroupsOnBranchAsync({
+      limit: paginatedQueryOptions.limit ?? UPDATE_GROUPS_LIMIT,
+      offset: paginatedQueryOptions.offset,
+      appId: projectId,
+      branchName,
+    });
+    renderUpdateGroupsAsTable(updateGroups, paginatedQueryOptions);
   } else {
     await paginatedQueryWithConfirmPromptAsync({
-      limit: options.limit ?? UPDATES_LIMIT,
-      offset: options.offset,
+      limit: paginatedQueryOptions.limit ?? UPDATE_GROUPS_LIMIT,
+      offset: paginatedQueryOptions.offset,
       queryToPerform: (limit, offset) =>
-        queryUpdateGroupsForBranchAsync(limit, offset, projectId, branchName),
+        queryUpdateGroupsOnBranchAsync({ limit, offset, appId: projectId, branchName }),
       promptOptions: {
         title: 'Load more branches?',
-        renderListItems: updates => renderUpdateGroups(updates, options, branchName),
+        renderListItems: updates => renderUpdateGroupsAsTable(updates, paginatedQueryOptions),
       },
     });
   }
 }
 
-async function queryUpdateGroupsForBranchAsync(
-  limit: number,
-  offset: number,
-  projectId: string,
-  branchName: string
-): Promise<BranchUpdateObject[]> {
-  const { app } = await UpdateQuery.viewBranchAsync({
-    appId: projectId,
-    name: branchName,
+export async function selectUpdateGroupOnBranchAsync({
+  projectId,
+  branchName,
+  paginatedQueryOptions: painatedQueryOptions,
+}: {
+  projectId: string;
+  branchName: string;
+  paginatedQueryOptions: PaginatedQueryOptions;
+}): Promise<BranchUpdateGroupObject> {
+  if (painatedQueryOptions.nonInteractive) {
+    throw new Error('Unable to select an update in non-interactive mode.');
+  } else {
+    const updateGroup = await paginatedQueryWithSelectPromptAsync({
+      limit: painatedQueryOptions.limit ?? UPDATE_GROUPS_LIMIT,
+      offset: painatedQueryOptions.offset,
+      queryToPerform: (limit, offset) =>
+        queryUpdateGroupsOnBranchAsync({ appId: projectId, branchName, limit, offset }),
+      promptOptions: {
+        title: 'Load more updates?',
+        createDisplayTextForSelectionPromptListItem: updateGroup =>
+          formatUpdateTitle(updateGroup[0]),
+        getIdentifierForQueryItem: updateGroup => updateGroup[0].group,
+      },
+    });
+
+    if (!updateGroup || updateGroup.length === 0) {
+      throw new Error(`Could not find any branches for project "${projectId}"`);
+    }
+
+    return updateGroup;
+  }
+}
```

Comment:
You can consider this as a nitpick but it seems those 3 could be refactored/simplified. They look mostly the same. You could create another, generic function and use it everywhere.

Reply:
happy to get into refactoring these methods, but if it's alright with you I'd prefer to do it in a follow-up once we've got pagination in all our commands.

Reply:
Sounds good.